src/esm_parser/esm_parser.py

```diff
@@ -2954,6 +3027,50 @@ def _config_init(self, user_config):
                     current_model=component,
                     user_config=user_config,
                     setup_config=setup_config)
+        # New organization of the standalone files
+        elif setup_name in self.config:
+            #import ipdb
+            #ipdb.set_trace()
+            standalone_model_config = self.config[setup_name]
+            setup_config["general"].update({"standalone": True})
+            setup_config["general"].update({"models": [standalone_model_config["model"]]})
+
+            resolve_choose_with_var(
+                "include_models",
+                standalone_model_config,
+                user_config=user_config,
+                setup_config=setup_config,
+            )
+
+            setup_config["general"]["include_models"] = standalone_model_config.get(
+                "include_models", []
+            )
+
+            #import ipdb
+            #ipdb.set_trace()
+
+            # If there is a key with name the coupled setup name, merge it with the
+            # general section and remove it
+            if coupled_setup and user_config["general"]["setup_name"] in user_config:
+                user_config["general"].update(
+                    user_config[user_config["general"]["setup_name"]]
+                )
+                del user_config[user_config["general"]["setup_name"]]
+
+            dict_merge(setup_config, self.config)
+
+            # TODO: this block below is absolutely nuts, remove it when cleaning
+            setup_config["general"]["valid_setup_names"] = valid_setup_names = list(
+                setup_config
+            )
+            setup_config["general"]["valid_setup_names"].remove(setup_name)
+            setup_config["general"]["valid_model_names"] = valid_model_names = [
+                setup_name
+            ]
```

Comment:
.....what?

First, take every key in the dictionary setup_config, as a list. Then, throw away the own name (since "awicm3" is actually "general"?) and then, hardwire model names to be just the setup name?

I though `"valid_model_names"` was supposed to be something like `["oifs", "fesom"]`

Reply:
Yeah I don't get those lines either, but changing the logic seems to create a lot of problems, so I leave the logic intact. We can address this when we refactor the parser (one day...)